lr-cli/cortx_setup/commands/storage/config.py

```diff
@@ -210,30 +238,88 @@ def run(self, **kwargs):
             'node_info_index',
             f'server_node>{self.machine_id}>type'
         )
+        if setup_type == None:
+            self.logger.error("Setup type is not set, please set the"
+            " setup type and try again, run following command to set"
+            " the setup type: 'cortx_config server config type <VM|HW>'"
+            )
+            raise RuntimeError("Could not find the setup type in conf store")
+
 
         if setup_type == "VM":
-            if name or storage_type or controller_type:
+            if (
+                name
+                or storage_type
+                or controller_type
+                or cvg
+                or data_devices
+                or metadata_devices
+            ):
                 if not self.enclosure_id:
                     self.enclosure_id = "enc_" + self.machine_id
                     self.refresh_key_map()
                     self.store_in_file()
-                    self.update_pillar_and_conf('enclosure_id', self.enclosure_id)
+                    self.update_pillar_and_conf(
+                        'enclosure_id',
+                        self.enclosure_id
+                    )
 
                 if name:
                     self.update_pillar_and_conf('name', name)
 
-                if storage_type:
+                if storage_type != "virtual":
+                    self.logger.error(
+                        "Storage type needs to be 'virtual' for VM")
+                    raise ValueError("Incorrect argument value provided")
+                else:
                     self.update_pillar_and_conf('storage_type', storage_type)
 
-                if controller_type:
-                    self.update_pillar_and_conf('controller_type', controller_type)
+                if controller_type != "virtual":
+                    self.logger.error(
+                        "Controller type needs to be 'virtual' for VM")
+                    raise ValueError("Incorrect argument value provided")
+                else:
+                    self.update_pillar_and_conf(
+                        'controller_type',
+                        controller_type
+                    )
+
+                if cvg > 1:
+                    self.logger.error(
+                        "The value of CVG has to be either 0 or 1")
+                    raise ValueError("Incorrect argument value provided")
+                else:
+                    self.update_pillar_and_conf('cvg', cvg)
+
+                if data_devices:
+                    ddevices = data_devices.split(",")
```

Comment:
When we use `nargs`, it will automatically be a list. This `split` will not work. 
```
>>> l = ['dev/mapper/mpatha']
>>> l.split(",")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'list' object has no attribute 'split'
```

Reply:
@83bhp  With the current code if user tries to provide comma separated values for data_devices and metadata_devices it is getting stored as single element in the list and hence split will not work as it is already list.
```
# cortx_setup storage config --cvg 1 --data_devices /dev/sdc,/dev/sdd --metadata_devices /dev/sdb          
data devices:  ['/dev/sdc,/dev/sdd']
metadata devices:  ['/dev/sdb']
```

If we use `nargs` then space separated values has to be provided by the user
```
#cortx_setup storage config --cvg 1 --data_devices /dev/sdc /dev/sdd --metadata_devices /dev/sdb          
data devices:  ['/dev/sdc', '/dev/sdd']
metadata devices:  ['/dev/sdb']
```

But if at all we want user to provide comma separated values then it's better to not use `nargs` and we can create list of devices using split.

Reply:
Addressed concerns from both of you, please review the code again.